x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/querydsl/query/SingleValueMatchQuery.java

```diff
@@ -46,15 +46,14 @@ public final class SingleValueMatchQuery extends Query {
      * This avoids reporting warnings when queries are not matching multi-values
      */
     private static final int MULTI_VALUE_MATCH_COST = 1000;
-    private static final IllegalArgumentException MULTI_VALUE_EXCEPTION = new IllegalArgumentException(
-        "single-value function encountered multi-value"
-    );
     private final IndexFieldData<?> fieldData;
     private final Warnings warnings;
+    private final IllegalArgumentException multiValueException;
 
-    public SingleValueMatchQuery(IndexFieldData<?> fieldData, Warnings warnings) {
+    public SingleValueMatchQuery(IndexFieldData<?> fieldData, Warnings warnings, String multiValueExceptionMessage) {
         this.fieldData = fieldData;
         this.warnings = warnings;
+        this.multiValueException = new IllegalArgumentException(multiValueExceptionMessage);
```

Comment:
Could we init this one lazily? It's pretty expensive to build it on every query.

Reply:
Removed the exception instantiation, and added a addWarning overload that just receives the class + message, for cases like this

Reply:
:+1: